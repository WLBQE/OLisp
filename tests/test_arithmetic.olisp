(define (int i) 1)
(define (double do) 2.1)

(print (i2d i))
(print (d2i do))

(print (+ (begin (print 1) 1) (begin (print 2) 2) (begin (print 3) 3)))

(define (double a) (+ -1.0 2.2))
(define (double b) 1.5)
(print (+ a b))
(print (* a b))
(print (- a b))

(define (int c) 25)
(define (int d) 10)
(define (int e) 1)
(print (* c d (+ e (+ e 1 1))))
(print (- c d))
(print (/ c d))
(print (% c d))
