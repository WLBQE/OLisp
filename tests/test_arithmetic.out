1
2
3
6
2.7
1.8
-0.3
1000
15
2
5
